### PR TITLE
HARMONY-1754: Change query-cmr to use the UMM-JSON granule format

### PR DESCRIPTION
**Pull request by ygliuvt**
## Jira Issue ID
HARMONY-1754

## Description
Change query-cmr to use the UMM-JSON granule format

## Local Test Steps
Build query-cmr docker image and run harmony locally, verify a harmony request works. e.g.
http://localhost:3000/C1234724470-POCLOUD/ogc-api-coverages/1.0.0/collections/all/coverage/rangeset?subset=lat(-45%3A45)&subset=lon(0%3A180)&maxResults=2&skipPreview=true&ignoreErrors=true
Also verify the stac catalogs and granule stac files output by query-cmr step have the correct content.

## PR Acceptance Checklist
* [x] Acceptance criteria met
* [x] Tests added/updated (if needed) and passing
* [ ] Documentation updated (if needed)

### CODE REVIEW

File: services/query-cmr/app/query.ts [hunk 1]
```diff
@@ -41,27 +41,37 @@ async function querySearchAfter(
...
   );
+
   const { hits } = cmrResponse;
   const newSearchAfter = cmrResponse.searchAfter;
   logger.info(`CMR Hits: ${hits}, Number of granules returned in this page: ${cmrResponse.granules.length}`);
   let totalItemsSize = 0;
   const outputItemSizes = [];
   const catalogs = cmrResponse.granules.map((granule) => {
-    const granuleSize = granule.granule_size ? parseFloat(granule.granule_size) : 0;
+    const archiveInfo = granule.umm.DataGranule?.ArchiveAndDistributionInformation;
```
Comment (chris-durbin) [inline services/query-cmr/app/query.ts:53]: This one is important for batching to make sure we obey size constraints. Is this returning the exact same thing as before or is it possibly different?

**Review by chris-durbin — COMMENTED** [review]
Looks good - I'll run a couple more tests tomorrow using HOSS and Concise to regression test additional behavior. So far I regression tested a large l2ss-py request and harmony-service-example using the new query-cmr and didn't see any kind of behavior change or performance degradation.

File: services/query-cmr/app/query.ts (hunk 1 above)
Comment (ygliuvt) [reply services/query-cmr/app/query.ts:53]: It should be exactly the same.

**Review by chris-durbin — APPROVED** [review]
Finished testing with Concise and HOSS and the behavior is the same.

**Review by indiejames — APPROVED** [review]

**Pull request merged**